### PR TITLE
Feature/code updates

### PR DESCRIPTION
**Pull request by achen2401**
-  add error checks, e.g. auth error, FHIR api error.  Currently, a blank screen is presented if there is any error.
-  fix `screening-selector.js` so only the required questionnaire & CQL json files are loaded based on selection. 
 Currently **ALL** questionnaire and CQL json files even for the questionnaire(s) that aren't selected
-  add ability to change survey theme and settings (see `/context/themes.js`, `/context/surveyjs.options.js`)
-  break up code in `SurveyComponent.vue` into smaller methods, remove duplicate code
-  add support for SASS; remove inline css and add separate stylesheet, see `app.scss`
-  add vuetify plugin (for additional UI components)

### CODE REVIEW

File: src/util/screening-selector.js [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-import elmJsonUsAudit from '../cql/UsAuditLogicLibrary.json';
-import elmJsonWhoAudit from '../cql/WhoAuditLogicLibrary.json';
-import elmJsonNidaQs from '../cql/NidaQsToUsAuditLogicLibrary.json';
 import valueSetJsonUsAudit from '../cql/valueset-db.json';
 
-
-export function getScreeningInstrument() {
-  let screeningInstrument = process.env.VUE_APP_ALCOHOL_SCREENING_INSTRUMENT.toLowerCase();
-
+//dynamically load questionnaire and cql JSON
```
Comment (mcjustin) [inline src/util/screening-selector.js:4]: @achen2401  I'd like to review the vue life cycle with you to better understand the scope here. I'm curious what this looks like if we're presenting a queue of questionnaires (I think the asbi system only does one).
For my reference: https://vuejs.org/guide/essentials/lifecycle.html#lifecycle-diagram

Comment (achen2401) [reply src/util/screening-selector.js:4]: @mcjustin yes, that is my understanding, too.  From what I have gathered,  the tool currently only presents one questionnaire at a time.  If we have a queue of questionnaires we will mostly likely need to refactor the code.  Also, it looks like the tool that transforms FHIR questionnaire into a SurveyJS json format also only supports **one** questionnaire.  I am looking at [here](https://github.com/asbi-cds-tools/questionnaire-to-survey/blob/751baa61f81314f7026ac5f7a5a34e03a80eeac5/fhirConversionTools.js#L8).

File: src/components/SurveyComponent.vue [hunk 2]
```diff
@@ -1,324 +1,277 @@
...
+    </v-alert>
   </div>
 </template>
 
 <script>
-
 import converter from 'questionnaire-to-survey';
 import { getScreeningInstrument } from '../util/screening-selector.js';
-import Worker from "../../node_modules/cql-worker/src/cql.worker.js"; // https://github.com/webpack-contrib/worker-loader
+import Worker from "../../node_modules/cql-worker/src/cql.worker.js"; //https://github.com/webpack-contrib/worker-loader
```
Comment (ivan-c) [inline src/components/SurveyComponent.vue:17]: Minor nitpick: removing a superfluous change to the original. I'm probably being overly paranoid, but might be conflicts/rebasing easier.
```suggestion
import Worker from "../../node_modules/cql-worker/src/cql.worker.js"; // https://github.com/webpack-contrib/worker-loader
```

Comment (ivan-c) [reply src/components/SurveyComponent.vue:17]: Also, try the **Commit suggestion** button if you haven't already!!

File: src/components/SurveyComponent.vue [hunk 3]
```diff
@@ -1,324 +1,277 @@
...
+        const [questionnaire, elmJson, valueSetJson] = data;
+        if (!questionnaire) throw Error("No questionnaire set");
+        self.questionnaire = questionnaire;
+        // Assemble the parameters needed by the CQL
+        let cqlParameters = {
+          DisplayScreeningScores: process.env.VUE_APP_DISPLAY_SCREENING_SCORES && process.env.VUE_APP_DISPLAY_SCREENING_SCORES.toLowerCase() == "true" ? true : false,
+          QuestionnaireURL: this.getQuestionnaireURL()
+        };
+        // Send the cqlWorker an initial message containing the ELM JSON representation of the CQL expressions
+        setupExecution(elmJson, valueSetJson, cqlParameters);
```
Comment (ivan-c) [inline src/components/SurveyComponent.vue:99]: Just curious- is this kinda like [`executeELM()` for the AHRQR app](https://github.com/uwcirg/AHRQ-CDS-Connect-PAIN-MANAGEMENT-SUMMARY/blob/develop/src/utils/executeELM.js#L17)?

File: src/components/SurveyComponent.vue [hunk 4]
```diff
@@ -1,324 +1,277 @@
...
+      if (questionnaireAuthor == 'practitioner') {
+        // Only add the `author` element if we can get the user id from the client
+        if (client.user && client.user.fhirUser) {
+          this.questionnaireResponse.author = {
+            reference: client.user.fhirUser
+          };
+        }
+      } else if (questionnaireAuthor == 'patient') {
+        this.questionnaireResponse.author = {
+          reference: this.questionnaireResponse.subject.reference
```
Comment (ivan-c) [inline src/components/SurveyComponent.vue:220]: Aside: I'm really surprised they have a config to control this. SoF should allow you to detect this information. Maybe they were in a non-compliant EHR?

File: src/components/SurveyComponent.vue [hunk 5]
```diff
@@ -1,324 +1,277 @@
...
+    },
+    initializeSurveyObjEvents() {
+      // Add an event listener which updates questionnaireResponse based upon user responses
+      this.survey.onValueChanging.add(function(sender, options) {
+        // We don't want to modify anything if the survey has been submitted/completed.
+        if (sender.isCompleted == true) return;
+        
+        if (options.value != null) {
+          // Find the index of this item (may not exist)
+          // NOTE: THIS WON'T WORK WITH QUESTIONNAIRES THAT HAVE NESTED ITEMS
```
Comment (ivan-c) [inline src/components/SurveyComponent.vue:232]: If we ever want to consider nested surveys or branching, this is slightly troubling, but probably something we can work around

File: src/components/SurveyComponent.vue [hunk 6]
```diff
@@ -1,324 +1,277 @@
...
-      if (qr) {
-        if (qr.resourceType == 'Bundle' && qr.entry) {
-          qr.entry.forEach(q => {
-            if (q.resource) patientBundle.entry.push({resource: q.resource});
+        // Write back to EHR only if `VUE_APP_WRITE_BACK_MODE` is set to 'smart'
+        if (process.env.VUE_APP_WRITE_BACK_MODE &&
+            process.env.VUE_APP_WRITE_BACK_MODE.toLowerCase() == 'smart') {
+          client.create(this.questionnaireResponse, {
+            headers: {
+              'Content-Type': 'application/fhir+json'
```
Comment (ivan-c) [inline src/components/SurveyComponent.vue:265]: We should start using this Content-Type!!

File: src/util/screening-selector.js (hunk 1 above)
Comment (ivan-c) [reply src/util/screening-selector.js:4]: Thanks for digging in Amy!

> If we have a queue of questionnaires we will mostly likely need to refactor the code

Another option could be to queue them in the EHR (launching the app for each survey), and launch for each Questionnaire. However, SoF may not be designed for that

**Review by ivan-c — APPROVED** [review]
Thank you Amy, this awesome!

**Comment by achen2401** [comment]
> @achen2401 Would upstream find any of these changes generally useful/helpful?
> 
> If we work more on this codebase (**if**), it would save us some trouble with conflicts to upstream our changes earlier than later.

@ivan-c yeah, especially error checking changes.  But I guess, at this point , it is not clear to me how much we will eventually diverge from the upstream codebase by any future changes we will need to make to fit our needs.

File: src/components/SurveyComponent.vue (hunk 3 above)
Comment (achen2401) [reply src/components/SurveyComponent.vue:99]: Yeah, I believe so.

File: src/util/screening-selector.js (hunk 1 above)
Comment (mcjustin) [reply src/util/screening-selector.js:4]: @ivan-c  re `Another option could be to queue them in the EHR`

I was thinking that a patient-specific CarePlan would be upstream of this scope (preferably stored in the EHR), with a PlanDefinition upstream of that. 

The asbi systems are a bit murky to me re actual implementation vs name-dropping:

No reference to "PlanDefinition" in the code, for neither the screener nor the intervention.
But, there is a json file for PlanDefinition in the *LogicFiles.zip for each of the 3 sub-audits, eg "PlanDefinition-AlcoholScreeningWhoAudit.json". 
These include a reference to "ApplyScreeningInstrument", which is defined in the CQL.

CarePlan
Only reference in the code is in asbi-screening-app/asbi-screening-app/src/cql/FHIRHelpers-4.0.1.json
eg:"name" : "{http://hl7.org/fhir}CarePlan[Status|Intent|ActivityStatus|ActivityIntent]"
No sample json files found for CarePlan

**Review by mcjustin — APPROVED** [review]
@achen2401 Sorry to be slow to approve this. Do you think you an merge this today? I'll be modifying some of these files for https://github.com/uwcirg/asbi-screening-app/pull/5

**Comment by achen2401** [comment]
@mcjustin I did merge the PR.  Let me know if you run into any issue though?

**Pull request merged**